### PR TITLE
upgrading go version to go 1.12

### PR DESCRIPTION
**Pull request by Avni-Sharma**

### CODE REVIEW

**Comment by Avni-Sharma** [comment]
Hi @davecheney I forgot to add the signoff in the commit message. Is there any other way in which I can do the same after I have raised a PR

**Comment by rosskukulinski** [comment]
@Avni-Sharma you should be able to amend your commit locally to include the signature then force-push to your branch.

**Comment by davecheney** [comment]
git commit -s —amend
git push -f

**Comment by davecheney** [comment]
Thank you. I will review this when the 0.11 window opens.

**Comment by davecheney** [comment]
@Avni-Sharma let's hold on this til Go 1.12.1 ships. There are some, uh, spicy bugs that are being backported from master rn.

**Comment by Avni-Sharma** [comment]
Sure

**Comment by davecheney** [comment]
I'm pretty sure 1.12 will be out soon, probably next week. Some of the reflection bugs are a bit scary.

**Comment by vaamarnath** [comment]
@Avni-Sharma Go 1.12.1 is out. @davecheney shouldn't we pick 1.12.1 to complete this PR?

**Comment by davecheney** [comment]
@Avni-Sharma yup, i'm just waiting for the 1.12.1 tag to show up in the docker hub, https://hub.docker.com/_/golang

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.11 AS build
+FROM golang:1.12 AS build
```
Comment (davecheney) [inline Dockerfile]: golang:1.12.1 when it becomes available.

**Pull request merged**